Lib/test/support/__init__.py

```diff
@@ -2873,3 +2874,40 @@ def __iter__(self):
         if self.iter_raises:
             1/0
         return self
+
+
+def in_systemd_nspawn() -> bool:
+    """
+    Test whether the test suite is running inside of systemd-nspawn container
+    """
+
+    try:
+        with open("/run/systemd/container", "rb") as fp:
+            return fp.read().rstrip() == b"systemd-nspawn"
+    except FileNotFoundError:
+        return False
+
+
+def in_systemd_nspawn_sync_suppressed() -> bool:
+    """
+    Test whether the test suite is runing in systemd-nspawn with ``--suppress-sync=true``
+
+    Return True if the test suite is being run inside a systemd-nspawn
+    container and ``--suppress-sync=true`` option is detected to be used,
+    False otherwise.  This can be used to skip tests that rely
+    on ``fsync()`` calls and similar not being intercepted.
+    """
+
+    if hasattr(os, "O_SYNC") and in_systemd_nspawn():
+        import errno
+
+        # If systemd-nspawn is used, O_SYNC flag will immediately trigger
+        # EINVAL.  Otherwise, ENOENT will be given instead.
+        try:
+            with os.open("", os.O_RDONLY | os.O_SYNC):
```

Comment:
It's strange to pass an empty path. I suggest to use ``__file__`` instead:

```suggestion
            with os.open(__file__, os.O_RDONLY | os.O_SYNC):
```